### ISSUE TITLE
Travis builds sometimes fail with timeout

### ISSUE THREAD

**Issue by bneumann** [opensheetmusicdisplay/opensheetmusicdisplay#287]
For instance: https://travis-ci.org/opensheetmusicdisplay/opensheetmusicdisplay/jobs/400113650

It seems the promise is not resolved in time also with a timeout of 4 seconds. This only happens (AFAIK) on the Travis build machines and mostly with Node V6. 
There might be an issue with our Promise implementation or the karma test "done" call.

**Comment by bneumann** [comment]
@sebastianhaas I put you on this because the two of us put in the most work recently

**Comment by matt-uib** [comment]
@bneumann please try with even bigger timeouts here in karma.conf.js:
// timeout in ms:
        browserNoActivityTimeout: 100000,
        captureTimeout: 60000,
I had to increase that in the past, as it never worked on my dev computer and then it did..
I guess loading a big piece takes too long.
Actually where are the 4 seconds? Please increase that to 60 seconds as first step.

**Comment by bneumann** [comment]
As discussed with @matt-uib: Increasing the timeout only helps us so far. We have a similar issue with memory leaks that are currently "fixed" by increasing the memory size in tests and using 64 bit only. We might end up with very specific machines running OSMD if we continue setting these boundary conditions :)

Anyways. I have a VM running with linux that shows both errors so I could start investigating

**Comment by bneumann** [comment]
I am testing again because all our tests seem to fail all the time. They work on my machine just fine and fail in my linux box. So I checked and the box only has 1 core. So I started stressing my CPU (smeels like burned dust now here) and run the tests again: Voilá failing hard!
So my guess is that we are on the edge of what the VM karma is running in can handle. So 2 possibilities:
1. We improve memory and perfomance (it started when we added loads of code so we can do a little boost here)
2. We find a way to make karma more tolerant to load peaks. 

Both ways are pretty hard to do :(
